src/swap/SwapReviewScreen.tsx

```diff
@@ -23,38 +22,33 @@ import TokenDisplay, { formatValueToDisplay } from 'src/components/TokenDisplay'
 import Touchable from 'src/components/Touchable'
 import InfoIcon from 'src/icons/InfoIcon'
 import { noHeader } from 'src/navigator/Headers'
-import { Screens } from 'src/navigator/Screens'
-import { StackParamList } from 'src/navigator/types'
 import DisconnectBanner from 'src/shared/DisconnectBanner'
 import colors from 'src/styles/colors'
 import fontStyles from 'src/styles/fonts'
 import { Spacing } from 'src/styles/styles'
 import variables from 'src/styles/variables'
-import { Field } from 'src/swap/useSwapQuote'
+import { swapUserInputSelector } from 'src/swap/selectors'
+import { swapStart } from 'src/swap/slice'
+import { Field, SwapInfo } from 'src/swap/types'
 import { coreTokensSelector } from 'src/tokens/selectors'
 import { divideByWei, multiplyByWei } from 'src/utils/formatting'
 import Logger from 'src/utils/Logger'
 import networkConfig from 'src/web3/networkConfig'
 import { walletAddressSelector } from 'src/web3/selectors'
 
-type Props = StackScreenProps<StackParamList, Screens.SwapReviewScreen>
-
 const TAG = 'SWAP_REVIEW_SCREEN'
-
-interface SwapInfo {
-  unvalidatedSwapTransaction: {
-    sellToken: string
-    buyToken: string
-    buyAmount: string
-    sellAmount: string
-    price: string
-    gas: string
-    gasPrice: string
+export function SwapReviewScreen() {
+  const initialUserInput = {
```

Comment:
i think this can be moved outside of the component to prevent it from being regenerated on every component rerender :)

Reply:
Moved in https://github.com/valora-inc/wallet/pull/2891/commits/da2c20e1a0837bdfd87830b0ca204ba6857b182d!